src/allmydata/frontends/magic_folder.py

```diff
@@ -483,50 +641,66 @@ def get_status(self):
             yield item
 
     def _get_filepath(self, relpath_u):
-        self._log("_get_filepath(%r)" % (relpath_u,))
         return extend_filepath(self._local_filepath, relpath_u.split(u"/"))
 
-    def _begin_processing(self, res):
-        self._log("starting processing loop")
-        self._processing = self._do_processing()
+    def stop(self):
+        """
+        Don't process queued items anymore.
+
+        :return Deferred: A ``Deferred`` that fires when processing has
+            completely stopped.
+        """
+        d = self._processing
+        self._processing_loop.stop()
+        self._processing = None
+        self._processing_loop = None
+        return d
+
+    def _begin_processing(self):
```

Comment:
In principle I could revert the change to the way the processing loop works now that we have `@inline_callbacks` to help Eliot out ... but I like the LoopingCall version more. :/  I won't press this point because it's not really central to the point here (make Magic-Folder logs easier to understand) but I hope it can just stay as it is in the branch now.